File: api/models/item.js
import mongoose from 'mongoose';

const ItemSchema = new mongoose.Schema({
  name: { type: String, required: true },
  quantity: { type: String, default: 1 },
  isPurchased: { type: Boolean, default: false },
  groupId: { type: String, ref: 'Group', required: true },
  addedBy: { type: String, ref: 'User', required: true },
  purchasedBy: { type: String, ref: 'User' },
  purchasedAt: { type: Date },
  deletedBy: { type: String, ref: 'User' },
  deletedAt: { type: Date },
  createdAt: { type: Date, default: Date.now }
}, { timestamps: true });

export default mongoose.model('Item', ItemSchema);

File: android/app/src/main/java/com/thaya/shop_family/models/Item.kt
package com.thaya.shop_family.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Item(
    @SerializedName("_id") val id: String,
    val name: String,
    val quantity: String,
    val isPurchased: Boolean,
    val groupId: String,
    val addedBy: String,
    val createdAt: String
) : Serializable
